File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/mapper/ParkingResponseMapper.java
package com.lattis.ellipse.data.network.model.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.data.network.model.response.parking.FindParkingDataResponse;
import com.lattis.ellipse.domain.model.Parking;

import javax.inject.Inject;


public class ParkingResponseMapper extends AbstractDataMapper<FindParkingDataResponse,Parking> {

    @Inject
    public ParkingResponseMapper() {
    }

    @NonNull
    @Override
    public Parking mapIn(@NonNull FindParkingDataResponse findParkingDataResponse) {

        Parking parking = new Parking();
        parking.setLongitude(findParkingDataResponse.getLongitude());
        parking.setParking_spot_id(findParkingDataResponse.getParking_spot_id());
        parking.setName(findParkingDataResponse.getName());
        parking.setDescription(findParkingDataResponse.getDescription());
        parking.setPic(findParkingDataResponse.getPic());
        parking.setType(findParkingDataResponse.getType());
        parking.setLatitude(findParkingDataResponse.getLatitude());
        parking.setParking_area_id(findParkingDataResponse.getParking_area_id());
        parking.setFleet_id(findParkingDataResponse.getFleet_id());
        parking.setOperator_id(findParkingDataResponse.getOperator_id());
        parking.setCustomer_id(findParkingDataResponse.getCustomer_id());
        return parking;
    }

    @NonNull
    @Override
    public FindParkingDataResponse mapOut(@NonNull Parking parking) {
        return null;
    }
}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/mapper/ParkingResponseMapper.kt
package com.lattis.data.mapper

import com.lattis.data.entity.response.parking.FindParkingDataResponse
import com.lattis.domain.models.Parking
import javax.inject.Inject

class ParkingResponseMapper @Inject constructor() :
    AbstractDataMapper<FindParkingDataResponse, Parking>() {
    override fun mapIn(findParkingDataResponse: FindParkingDataResponse?): Parking {
        val parking = Parking()
        parking.longitude = findParkingDataResponse?.longitude
        parking.parking_spot_id = findParkingDataResponse?.parking_spot_id
        parking.name = findParkingDataResponse?.name
        parking.description = findParkingDataResponse?.description
        parking.pic = findParkingDataResponse?.pic
        parking.type = findParkingDataResponse?.type
        parking.latitude = findParkingDataResponse?.latitude
        parking.parking_area_id = findParkingDataResponse?.parking_area_id
        parking.fleet_id = findParkingDataResponse?.fleet_id
        parking.operator_id = findParkingDataResponse?.operator_id
        parking.customer_id = findParkingDataResponse?.customer_id
        return parking
    }

    override fun mapOut(parking: Parking?): FindParkingDataResponse? {
        return null
    }
}